specification/web/resource-manager/Microsoft.Web/stable/2023-01-01/CommonDefinitions.json

```diff
@@ -2982,6 +2982,10 @@
               "description": "client certificate authentication comma-separated exclusion paths",
               "type": "string"
             },
+            "endToEndEncryptionEnabled": {
```

Comment:
>endToEndEncryptionEnabled [](http://example.com/codeflow?start=13&length=25)

Recommend enum over boolean, e.g. encryption: { EndToEnd, None }

Reply:
The feature was already written as a boolean flag, though. I don't think changing it here makes sense.

Reply:
The ARM feedback is to use enum. Already written doesn't really matter. 

Generally enums make better properties than booleans. They are more descriptive and flexible while being easier for customers to discover, understand and use. Ref: [https://eng.ms/docs/products/arm/rp_onboarding/process/api_review_best_practices](https://eng.ms/docs/products/arm/rp_onboarding/process/api_review_best_practices). They allow values other than just true/false and support description strings. Even if you only ever want two values, you can have a string enum with True/False (or Enabled/Disabled, etc.).

It's not required: just highly recommended so I won't hold ARM approval for this, but please seriously consider making the change.